src/ipc.rs

```diff
@@ -570,6 +570,24 @@ impl Deref for IpcSharedMemory {
     }
 }
 
+impl IpcSharedMemory {
+    /// Returns a mutable reference to the deref of this [`IpcSharedMemory`].
+    ///
+    /// # Safety
+    ///
+    /// This is safe if there is only one reader/writer on the data.
+    /// User can achieve this by not cloning [`IpcSharedMemory`]
+    /// and serializing/deserializing only once.
+    #[inline]
+    pub unsafe fn deref_mut(&mut self) -> &mut [u8] {
```

Comment:
I'm a bit confused on how this works. Do we know that if this code is running, then the other process is guaranteed to not be be able to access this memory?

Reply:
no there is no such guarantee, hence the method is marked unsafe (we will probably need named locks and serialization that takes ownership)

Reply:
users needs to make sure that only one process/thread uses memory (no synchronization is guaranteed as user must send ipcsharedmem back), this works if this shared memory is used for something like req-resp, for example webgpu's mapasync where we map (clone) buffer to JS thread (while it's mapped GPU will not touch it) then when unmaping we send this data back.

Reply:
I update PR description with example of usage (there is also bench in PR that can act as example).